components/StateMultipleChoiceQuizPage.tsx

```diff
@@ -1,10 +1,55 @@
-import React from "react";
-import { Text, View } from "react-native";
+import React, { useEffect, useState } from "react";
+import { Text, View, Button } from "react-native";
+import { Question } from "../types/Question";
+import { quizQuestions } from "../types/Question";
+
+export default function StateMultipleChoiceQuizApp() {
+  const [currentQuestionIndex, setCurrentQuestion] = useState<number>(0);
+  const [score, setScore] = useState<number>(0);
+  const [quizFinished, setQuizFinished] = useState<boolean>(false);
+  const [shuffledQuestions, setShuffledQuestions] = useState<Question[]>([]);
+
+  useEffect(() => {
+    const shuffled = [...quizQuestions].sort(() => Math.random() - 0.5);
+    setShuffledQuestions(shuffled);
+  }, []);
+  const currentQuestion: Question = shuffledQuestions[currentQuestionIndex] || {
+    questionText: "",
+    option: [],
+    correctAnswer: "",
+  };
+  const handleAnswer = (selectedOption: string) => {
+    if (selectedOption === currentQuestion.correctAnswer) {
+      setScore(score + 1);
+    }
+    if (currentQuestionIndex < quizQuestions.length - 1) {
+      setCurrentQuestion(currentQuestionIndex + 1);
+    } else {
+      setQuizFinished(true);
+    }
+  };
 
```

Comment:
Rearrange these properly.
```suggestion
  const [quizFinished, setQuizFinished] = useState<boolean>(false);
  const [shuffledQuestions, setShuffledQuestions] = useState<Question[]>([]);
  const currentQuestion: Question = shuffledQuestions[currentQuestionIndex] || {
    questionText: "",
    option: [],
    correctAnswer: "",
  };

  useEffect(() => {
    const shuffled = [...quizQuestions].sort(() => Math.random() - 0.5);
    setShuffledQuestions(shuffled);
  }, []);

  const handleAnswer = (selectedOption: string) => {
    if (selectedOption === currentQuestion.correctAnswer) {
      setScore(score + 1);
    }
    if (currentQuestionIndex < quizQuestions.length - 1) {
      setCurrentQuestion(currentQuestionIndex + 1);
    } else {
      setQuizFinished(true);
    }
  };
  
```

Reply:
I have in the recent commit, pls check